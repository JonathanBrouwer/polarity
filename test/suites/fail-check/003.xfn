codata Bool : Type {}

impl Bool {
    codef True(): Bool {}
    codef False(): Bool {}
}

data Nat : Type {
    Z: Nat,
    S(n: Nat) : Nat,
}

data Foo(b: Bool) : Type {
    Bar : Foo(True),
    Baz : Foo(False),
}

impl Foo {
    def Foo(True).foo() : Nat {
        Bar() => Z,
        Baz() absurd,
    }
}

Baz.foo()
