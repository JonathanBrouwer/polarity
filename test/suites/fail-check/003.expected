The following terms are not equal:
  1: Foo(False)
  2: Foo(True)
